### PR TITLE
robustness improvements to update_problem! and maxtimetocharge_discharge

### PR DESCRIPTION
**Pull request by sinnott**
Code changes to protect against cases where storage or generator storage devices have maxcharge or maxdischarge equal to 0 (which were resulting in Inf or Nan, causing assess() to fail).

### CODE REVIEW

File: src/simulations/modern/Modern.jl [hunk 1]
```diff
@@ -29,7 +29,7 @@ function assess(
     for _ in 1:threads
         @spawn assess(simspec, resultspec, system, samples, results)
     end
-
+
```
Comment (GordStephen) [inline src/simulations/modern/Modern.jl]: Just a nitpick, but can you eliminate this extra space?

File: src/simulations/modern/Modern.jl [hunk 2]
```diff
@@ -65,7 +65,6 @@ function assess(
     # TODO: Implement simulation-level RNG (Random123?)
     rng = GLOBAL_RNG
     for (s, _) in samples
-
```
Comment (GordStephen) [inline src/simulations/modern/Modern.jl]: Not sure how these lines got removed (here and below), but can you add them back?

File: src/simulations/modern/Modern.jl (hunk 1 above)
Comment (sinnott) [reply src/simulations/modern/Modern.jl]: Done in today's commit.

File: src/simulations/modern/Modern.jl (hunk 2 above)
Comment (sinnott) [reply src/simulations/modern/Modern.jl]: Done in today's commit.

**Pull request merged**